### DISCUSSION TITLE
How to use a pre-trained model in model zoo as a sub-block in my own model?

### DISCUSSION THREAD

**Discussion by ybz79** [apache/incubator-mxnet#16227] [Q&A] [answered]
## Description
I want to use a pretrained model in the model zoo (such as BERT) as a layer in my own model. 
```python
   class Mymodel(nn.Block):
        def __init__(bert_model):
               self.bert = bert_model
               .....

  
  bert_model = get_model(....)
  model = Mymodel(bert_model)
```

But I have a small problem about the naming, since bert_model and other parameters in my own model didn't have a same prefix, which can cause error in model.save_params()
, how can I slove this problem?

**Comment by zachgk** [comment, marked as answer]
You should use Mymodel.name_scope() when defining the child blocks. Take a look at the documentation in http://beta.mxnet.io/api/gluon/mxnet.gluon.nn.Block.html. Also, use model.save_parameters as save_params is deprecated.
